WebHostLib/tracker.py

```diff
@@ -11,9 +11,48 @@
 from WebHostLib import app, cache, Room
 from Utils import restricted_loads
 from worlds import lookup_any_item_id_to_name, lookup_any_location_id_to_name
+from worlds.AutoWorld import AutoWorldRegister
 from MultiServer import get_item_name_from_id, Context
 from NetUtils import SlotType
 
+
+class PlayerTracker:
+
+    template: str = 'playerTracker.html'
+    icons: Dict[str, str] = {}
+    regions: Dict[str, List[str]] = {}
+    checks_done: Dict[str, Set[str]] = {}
+    room: Room
+    all_locations: Set[str]
+    all_prog_items: Counter[str]
+    team: int
+    player: int
+    name: str
+    checked_locations: Set[str]
+    received_items: Counter[str]
+    slot_data = [Dict[any, any]]
+    theme: str
+    prog_items: Counter[str]
```

Comment:
These are defined as class variables, not necessarily as instance variables. Is this intentional?

Reply:
These are all defined outside the init in order to support IDE typing and the ones that aren't created in the instance are optional

Reply:
These are defined as class variables meaning they belong to the class object itself (the type) and _not_ any instances. It doesn't necessarily enable typing. You could set them all in the \_\_init\_\_ to reasonable defaults, with typing info, and get the same result but without mangling variable semantics.